app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala

```diff
@@ -49,96 +51,98 @@ class EditSetRecordController @Inject() (
   editSetRecordEdit: editSetRecordEdit,
   editSetRecordEditDiscard: editSetRecordEditDiscard,
   editSetRecordEditSave: editSetRecordEditSave
-) extends MessagesAbstractController(messagesControllerComponents) with I18nSupport with Secured with FormSupport {
+)(implicit val ec: ExecutionContext)
+    extends MessagesAbstractController(messagesControllerComponents) with I18nSupport with Secured with FormSupport {
 
   private val logger: Logger = Logger(this.getClass)
   private lazy val legalStatuses: Seq[LegalStatus] = referenceDataService.getLegalStatuses
   private lazy val creators: Seq[Creator] = referenceDataService.getCreators
   private lazy val placesOfDeposit: Seq[PlaceOfDeposit] = referenceDataService.getPlacesOfDeposit
 
-  def viewEditRecordForm(id: String, recordId: String): Action[AnyContent] = Action {
+  def viewEditRecordForm(id: String, recordId: String): Action[AnyContent] = Action.async {
     implicit request: Request[AnyContent] =>
-      withUser { user =>
+      withUserAsync { user =>
         logger.info(s"The edit set id is $id for record id $recordId")
         val editSetName = editSets.getEditSet().name
```

Comment:
I've raised [another ticket](https://national-archives.atlassian.net/browse/PACT-806) to address this. We don't want to be calling the API to get the whole edit set just to have a name to put into a back link.